crates/subspace-archiving/src/archiver.rs

```diff
@@ -185,6 +185,33 @@ pub struct NewArchivedSegment {
     pub object_mapping: Vec<PieceObjectMapping>,
 }
 
+impl NewArchivedSegment {
+    /// Returns all the object mappings in this archived segment.
+    //
+    // TODO: make this a double-ended lazy iterator, for efficiency
+    pub fn object_mappings(&self) -> Vec<ObjectMapping> {
+        // Iterate through the object mapping vector for each piece
+        self.object_mapping
+            .iter()
+            .enumerate()
+            .flat_map(|(piece_position, mappings)| {
+                // And then through each individual object mapping in the piece
+                mappings
+                    .objects
+                    .iter()
+                    .map(move |piece_object| ObjectMapping::V0 {
+                        object_id: piece_object.hash(),
```

Comment:
This property is called `hash` throughout, I'd probably keep it that way or rename consistently